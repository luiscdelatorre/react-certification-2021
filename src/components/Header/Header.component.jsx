import React from 'react';
import { IoLogoYoutube } from 'react-icons/io5';

import Searchbar from '../Searchbar';
import User from '../User';
import {
  HeaderCenter,
  HeaderContainer,
  HeaderLeft,
  HeaderRight,
  Logo,
  MenuHamburger,
} from './Header.styles';

const Header = ({ isMenuExpanded, onToggleMenu }) => {
  const toggleMenu = () => {
    onToggleMenu();
  };

  return (
    <HeaderContainer data-testid="header">
      <HeaderLeft>
        <MenuHamburger
          isMenuExpanded={isMenuExpanded}
          onClick={toggleMenu}
          data-testid="button-menu"
        />
        <Logo to="/">
          <IoLogoYoutube />
          <span>YuTuv</span>
        </Logo>
      </HeaderLeft>
      <HeaderCenter>
        <Searchbar />
      </HeaderCenter>
      <HeaderRight>
        <User />
      </HeaderRight>
    </HeaderContainer>
  );
};

export default Header;
