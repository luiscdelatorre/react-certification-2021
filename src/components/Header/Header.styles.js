import { Link } from 'react-router-dom';
import styled from 'styled-components';
import device from '../../config/device';

const Header = styled.header`
  grid-area: header;
  background-color: ${(props) => props.theme.menuBackground};
  display: flex;
  justify-content: space-between;
  padding: 2rem;
  border-bottom: 1px solid ${(props) => props.theme.borderColor};
`;

const HeaderItem = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
`;

const HeaderLeft = styled(HeaderItem)`
  justify-content: flex-start;
  @media ${device.laptop} {
    flex-basis: 14rem;
  }
`;

const HeaderCenter = styled(HeaderItem)`
  justify-content: flex-end;
  flex: 1;
  @media ${device.laptop} {
    justify-content: center;
  }
`;

const HeaderRight = styled(HeaderItem)`
  justify-content: flex-end;
  @media ${device.laptop} {
    flex-basis: 8rem;
  }
`;

const MenuHamburger = styled.button`
  position: relative;
  background: none;
  display: none;
  padding: 1.5rem 1rem;
  height: 4rem;
  width: 5rem;
  @media ${device.laptop} {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
  }
  &:focus,
  &:hover {
    outline: none;
    box-shadow: none;
    &::before,
    &::after {
      background-color: ${(props) => props.theme.menuButtonActive};
    }
  }
  &::before,
  &::after {
    content: '';
    transition: ${(props) => props.theme.transitionFast};
    background-color: ${(props) => props.theme.menuButton};
    width: ${(props) => (props.isMenuExpanded ? '3rem' : '2rem')};
    display: block;
    height: 0.3rem;
  }
  &:active {
    animation-name: '';
  }
`;

const Logo = styled(Link)`
  color: inherit;
  display: inline-flex;
  margin: 0 1rem;
  text-decoration: none;
  align-items: center;
  font-family: 'Cabin Condensed', sans-serif;
  position: relative;
  span {
    font-size: 3.3rem;
  }
  svg {
    color: ${(props) => props.theme.logoPrimary};
    font-size: 3.5rem;
    margin-right: 0.3rem;
    z-index: 1;
  }
  &::before {
    content: '';
    background-color: ${(props) => props.theme.logoSecondary};
    height: 2rem;
    width: 2rem;
    position: absolute;
    top: 1rem;
    left: 1rem;
    z-index: 0;
  }
  &::after {
    content: 'React';
    display: block;
    color: ${(props) => props.theme.textSecondary};
    font-size: 1.2rem;
    position: absolute;
    top: 0.1rem;
    right: 0;
  }
`;

const Styled = {
  Header,
  HeaderLeft,
  HeaderRight,
  HeaderCenter,
  Logo,
  MenuHamburger,
};
export default Styled;
