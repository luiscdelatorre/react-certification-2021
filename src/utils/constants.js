const AUTH_STORAGE_KEY = 'wa_cert_authenticated';
const SEARCH_STORAGE_KEY = 'wa_cert_search';
const API_URL = `https://www.googleapis.com/youtube/v3`;

export { AUTH_STORAGE_KEY, SEARCH_STORAGE_KEY, API_URL };
