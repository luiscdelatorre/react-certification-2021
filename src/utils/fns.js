const random = (limit) => {
  return Math.floor(Math.random() * limit);
};

const formatDate = (dateString) => {
  const date = new Date(dateString);
  return date.toLocaleDateString();
};

export { random, formatDate };
