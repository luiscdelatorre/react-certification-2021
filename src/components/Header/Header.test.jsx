import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AuthProvider from '../../providers/Auth';
import SessionDataProvider from '../../providers/SessionData/SessionData.provider';
import SearchProvider from '../../providers/Search';
import Header from './Header.component';
import { storage } from '../../utils/storage';
import { MENU_STORAGE_KEY } from '../../utils/constants';

describe('Header component', () => {
  it('Should expand menu size on menu button click', () => {
    storage.set(MENU_STORAGE_KEY, 'compact');
    const { rerender } = render(
      <BrowserRouter>
        <AuthProvider>
          <SearchProvider>
            <SessionDataProvider>
              <Header />
            </SessionDataProvider>
          </SearchProvider>
        </AuthProvider>
      </BrowserRouter>
    );

    const button = screen.queryByTestId('button-menu');
    const initialClassName = button.className;
    fireEvent.click(button);
    rerender(
      <BrowserRouter>
        <AuthProvider>
          <SearchProvider>
            <SessionDataProvider>
              <Header />
            </SessionDataProvider>
          </SearchProvider>
        </AuthProvider>
      </BrowserRouter>
    );
    expect(initialClassName).not.toBe(button.className);
    expect(storage.get(MENU_STORAGE_KEY)).not.toBe('compact');
  });

  it('Validates Header snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <AuthProvider>
          <SearchProvider>
            <SessionDataProvider>
              <Header />
            </SessionDataProvider>
          </SearchProvider>
        </AuthProvider>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
