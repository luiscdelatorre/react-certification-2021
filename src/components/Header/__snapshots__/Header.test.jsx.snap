// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Validates Header snapshot 1`] = `
<header
  className="sc-bqyKva llbqAk"
  data-testid="header"
>
  <div
    className="sc-kstrdz sc-hBEYos jqiJcG iZrsmI"
  >
    <button
      className="sc-bkzZxe deHug"
      data-testid="button-menu"
      onClick={[Function]}
    />
    <a
      className="sc-idOhPF gCEokI"
      href="/"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M508.64 148.79c0-45-33.1-81.2-74-81.2C379.24 65 322.74 64 265 64h-18c-57.6 0-114.2 1-169.6 3.6C36.6 67.6 3.5 104 3.5 149 1 184.59-.06 220.19 0 255.79q-.15 53.4 3.4 106.9c0 45 33.1 81.5 73.9 81.5 58.2 2.7 117.9 3.9 178.6 3.8q91.2.3 178.6-3.8c40.9 0 74-36.5 74-81.5 2.4-35.7 3.5-71.3 3.4-107q.34-53.4-3.26-106.9zM207 353.89v-196.5l145 98.2z"
        />
      </svg>
      <span>
        YuTuv
      </span>
    </a>
  </div>
  <div
    className="sc-kstrdz sc-fodVxV jqiJcG dnhjTT"
  >
    <div
      className="sc-bdfBwQ gmamkS"
      data-testid="searchbar"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M221.09 64a157.09 157.09 0 10157.09 157.09A157.1 157.1 0 00221.09 64z"
          fill="none"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
        <path
          d="M338.29 338.29L448 448"
          fill="none"
          strokeLinecap="round"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
      </svg>
      <form
        data-testid="form-search"
        onSubmit={[Function]}
      >
        <input
          className="sc-gsTCUz eSCwiF"
          data-testid="input-search"
          onBlur={[Function]}
          placeholder="Search"
          type="search"
        />
      </form>
    </div>
  </div>
  <div
    className="sc-kstrdz sc-fFubgz jqiJcG bBSCrS"
  >
    <button
      className="sc-eCssSg byGTwv"
      data-testid="user-default-icon"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm-50.22 116.82C218.45 151.39 236.28 144 256 144s37.39 7.44 50.11 20.94c12.89 13.68 19.16 32.06 17.68 51.82C320.83 256 290.43 288 256 288s-64.89-32-67.79-71.25c-1.47-19.92 4.79-38.36 17.57-51.93zM256 432a175.49 175.49 0 01-126-53.22 122.91 122.91 0 0135.14-33.44C190.63 329 222.89 320 256 320s65.37 9 90.83 25.34A122.87 122.87 0 01382 378.78 175.45 175.45 0 01256 432z"
        />
      </svg>
    </button>
  </div>
</header>
`;
